iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveTableUtil.java

```diff
@@ -182,24 +183,41 @@ private static List<DataFile> getDataFiles(RemoteIterator<LocatedFileStatus> fil
     return dataFiles;
   }
 
-  public static void appendFiles(URI fromURI, String format, Table icebergTbl, boolean isOverwrite, Configuration conf)
-      throws SemanticException {
+  public static void appendFiles(URI fromURI, String format, Table icebergTbl, boolean isOverwrite,
+      Map<String, String> partitionSpec, Configuration conf) throws SemanticException {
     try {
       Transaction transaction = icebergTbl.newTransaction();
       if (isOverwrite) {
         DeleteFiles delete = transaction.newDelete();
-        delete.deleteFromRowFilter(Expressions.alwaysTrue());
+        if (partitionSpec != null) {
+          for (Map.Entry<String, String> part : partitionSpec.entrySet()) {
+            delete.deleteFromRowFilter(Expressions.equal(part.getKey(),
+                Conversions.fromPartitionString(icebergTbl.schema().findType(part.getKey()), part.getValue())));
```

Comment:
could we extract this to partKey var?
icebergTbl.schema().findType(part.getKey()